<a href="https://colab.research.google.com/github/DalpesCS50/trab_estagio4_health-insurance-claims-prediction/blob/main/notebooks/01_exploracao_inicial_health_insurance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Predição do Valor de Sinistros em Seguro Saúde

## Estágio Supervisionado IV - Matemática Atuarial III

### Exploração inicial do conjunto de dados Health Insurance

**Objetivo:** realizar a exploração inicial do conjunto de dados *Health Insurance Data Set*, identificando sua estrutura, suas variáveis e as principais características dos dados que serão utilizados no desenvolvimento de um modelo preditivo para estimar o valor dos sinistros em seguro saúde.

## 1. Definição do problema

O custo dos sinistros é uma variável fundamental para a gestão de uma carteira de seguros, pois influencia diretamente a precificação, o planejamento financeiro e a avaliação dos riscos assumidos pela seguradora.

Neste projeto, pretende-se investigar se as características observadas dos segurados podem ser utilizadas para prever o valor dos sinistros em seguro saúde.

### Pergunta preditiva

**É possível estimar o valor do sinistro de um segurado a partir de suas características demográficas e de seus fatores de risco relacionados à saúde?**

### Variável resposta

A variável que se pretende prever é **`claim`**, que representa o valor do sinistro reclamado pelo segurado.

### Variáveis preditoras

As demais características disponíveis no conjunto de dados serão inicialmente consideradas como potenciais variáveis explicativas do valor dos sinistros. A relevância de cada uma delas será investigada posteriormente durante a análise exploratória e a modelagem.

In [1]:
import pandas as pd

In [2]:
url = "https://raw.githubusercontent.com/DalpesCS50/trab_estagio4_health-insurance-claims-prediction/refs/heads/main/data/healthinsurance.csv"

dados = pd.read_csv(url)

In [3]:
dados.head()

,age,sex,weight,bmi,hereditary_diseases,no_of_dependents,smoker,city,bloodpressure,diabetes,regular_ex,job_title,claim
0,60.0,male,64,24.3,NoDisease,1,0,NewYork,72,0,0,Actor,13112.6
1,49.0,female,75,22.6,NoDisease,1,0,Boston,78,1,1,Engineer,9567.0
2,32.0,female,64,17.8,Epilepsy,2,1,Phildelphia,88,1,1,Academician,32734.2
3,61.0,female,53,36.4,NoDisease,1,1,Pittsburg,72,1,0,Chef,48517.6
4,19.0,female,50,20.6,NoDisease,0,0,Buffalo,82,1,0,HomeMakers,1731.7


In [4]:
dados.shape

(15000, 13)

In [5]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   age                  14604 non-null  float64
 1   sex                  15000 non-null  object 
 2   weight               15000 non-null  int64  
 3   bmi                  14044 non-null  float64
 4   hereditary_diseases  15000 non-null  object 
 5   no_of_dependents     15000 non-null  int64  
 6   smoker               15000 non-null  int64  
 7   city                 15000 non-null  object 
 8   bloodpressure        15000 non-null  int64  
 9   diabetes             15000 non-null  int64  
 10  regular_ex           15000 non-null  int64  
 11  job_title            15000 non-null  object 
 12  claim                15000 non-null  float64
dtypes: float64(3), int64(6), object(4)
memory usage: 1.5+ MB


In [6]:
dados.isnull().sum()

,0
age,396
sex,0
weight,0
bmi,956
hereditary_diseases,0
no_of_dependents,0
smoker,0
city,0
bloodpressure,0
diabetes,0


In [7]:
(dados.isnull().sum() / len(dados)) * 100

,0
age,2.640000
sex,0.000000
weight,0.000000
bmi,6.373333
hereditary_diseases,0.000000
no_of_dependents,0.000000
smoker,0.000000
city,0.000000
bloodpressure,0.000000
diabetes,0.000000


In [8]:
dados.isnull().any(axis=1).sum()

np.int64(1352)

In [9]:
dados[dados["age"].isnull() & dados["bmi"].isnull()].shape

(0, 13)

In [10]:
dados.duplicated().sum()

np.int64(1096)

In [11]:
dados[dados.duplicated(keep=False)].sort_values(
    by=list(dados.columns)
).head(20)

,age,sex,weight,bmi,hereditary_diseases,no_of_dependents,smoker,city,bloodpressure,diabetes,regular_ex,job_title,claim
2005,18.0,female,40,24.8,NoDisease,0,0,Charleston,60,0,0,HouseKeeper,2203.7
2362,18.0,female,40,24.8,NoDisease,0,0,Charleston,60,0,0,HouseKeeper,2203.7
4078,18.0,female,44,26.3,NoDisease,0,0,Fargo,76,1,0,Labourer,2198.2
8072,18.0,female,44,26.3,NoDisease,0,0,Fargo,76,1,0,Labourer,2198.2
833,18.0,female,44,NaN,NoDisease,0,0,Charleston,96,1,0,Photographer,21344.9
6897,18.0,female,44,NaN,NoDisease,0,0,Charleston,96,1,0,Photographer,21344.9
1377,18.0,female,46,29.2,NoDisease,0,0,SantaRosa,64,1,1,FashionDesigner,7323.7
10350,18.0,female,46,29.2,NoDisease,0,0,SantaRosa,64,1,1,FashionDesigner,7323.7
9571,18.0,female,48,20.3,NoDisease,0,0,Nashville,60,1,0,HomeMakers,11482.6
12142,18.0,female,48,20.3,NoDisease,0,0,Nashville,60,1,0,HomeMakers,11482.6


In [12]:
dados.drop_duplicates().shape

(13904, 13)

### 1.1 Verificação inicial da qualidade dos dados

O conjunto de dados original possui **15.000 observações e 13 variáveis**.

A análise de valores ausentes identificou:

- **396 valores ausentes em `age`**, correspondentes a **2,64%** das observações;
- **956 valores ausentes em `bmi`**, correspondentes a aproximadamente **6,37%** das observações;
- não foram identificados valores ausentes nas demais variáveis;
- não existem registros em que `age` e `bmi` estejam simultaneamente ausentes.

No total, **1.352 observações apresentam pelo menos um valor ausente**, enquanto **13.648 observações estão completas em todas as variáveis**.

Também foram identificadas **1.096 linhas duplicadas** considerando conjuntamente as 13 variáveis do conjunto de dados. Caso as duplicidades sejam removidas, a base passa de 15.000 para **13.904 registros únicos**.

Nesta etapa exploratória, os valores ausentes e os registros duplicados foram apenas identificados. As decisões relativas ao tratamento dessas ocorrências serão realizadas posteriormente, durante a etapa de preparação dos dados para a modelagem.